Tools/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,197 @@
+param (
+    [switch]$IncludeSensitiveData
+)
+
+function Show-Warning {
+    param (
+        [string]$Message
+    )
+    Write-Warning $Message
+}
+
+function Get-SystemConfiguration {
+    Write-Host "Collecting system configuration..."
+    try {
+        Start-Process -FilePath "msinfo32.exe" -ArgumentList "/report", "msinfo32.txt" -Wait
+        Write-Host "msinfo32 completed."
+    } catch {
+        Show-Warning "Failed to collect system configuration: $_"
+    }
+}
+
+function Get-EventLogs {
+    Write-Host "Collecting event logs..."
+    try {
+        $logTypes = @("system", "security", "application")
+        foreach ($logType in $logTypes) {
+            wevtutil epl $logType "$logType.evtx"
+            wevtutil al "$logType.evtx"
+        }
+        Write-Host "Event logs collection completed."
+    } catch {
+        Show-Warning "Failed to collect event logs: $_"
+    }
+}
+
+function Get-DriversList {
+    Write-Host "Collecting drivers list..."
+    try {
+        driverquery /v /fo CSV > drv_list.csv
+        Write-Host "Drivers list collection completed."
+    } catch {
+        Show-Warning "Failed to collect drivers list: $_"
+    }
+}
+
+function Get-VirtioWinStorageDrivers {
+    Write-Host "Collecting virtio-win storage drivers registry info..."
+    try {
+        $registryPaths = @(
+            "HKLM\SYSTEM\CurrentControlSet\Services\Disk /v IoTimeoutValue",
+            "HKLM\SYSTEM\CurrentControlSet\Services\Disk /v TimeoutValue",
+            "HKLM\SYSTEM\CurrentControlSet\Services\viostor\Parameters /v IoTimeoutValue",
+            "HKLM\SYSTEM\CurrentControlSet\Services\vioscsi\Parameters /v IoTimeoutValue"
+        )
+        foreach ($path in $registryPaths) {
+            reg query $path | Out-File -FilePath virtio_disk.txt -Append

```

Comment:
I am wondering if it doesn't make more sense to use try/catch on single value . If we fail to read the first value, we might be able to read others.

Reply:
Also, it is better to use the Get-ItemProperty API instead of calling some application